### ISSUE TITLE
Sticky header does not work  if there is a view above RecyclerView

### ISSUE THREAD

**Issue by iloveaman** [bgogetap/StickyHeaders#9]
fix this bug by add those code below:
StickyLayoutManager.java

 
```
   RecyclerView recyclerView;
    @Override
    public void onAttachedToWindow(RecyclerView view) {
        super.onAttachedToWindow(view);
        recyclerView = view;
    }
 
```

ViewRetriever.java
```
final class RecyclerViewRetriever implements ViewRetriever {

        private RecyclerView.Recycler recycler;
        private RecyclerView recyclerView;

        RecyclerViewRetriever() {
        }

        RecyclerViewRetriever setRecycler(RecyclerView.Recycler recycler, RecyclerView recyclerView) {
            this.recycler = recycler;
            this.recyclerView = recyclerView;
            return this;
        }

        @Override
        public View getViewForPosition(int position) {
            View view = recycler.getViewForPosition(position);
            int[] positions = new int[2];
            recyclerView.getLocationInWindow(positions);
            Rect rectF = new Rect();
            recyclerView.getLocalVisibleRect(rectF);
            int[] positions2 = new int[2];
            recyclerView.getLocationOnScreen(positions2);
            ViewGroup.MarginLayoutParams marginLayoutParams = (ViewGroup.MarginLayoutParams) view.getLayoutParams();
            marginLayoutParams.topMargin = positions[1] - DecorUtils.getStratusBarHeight(view.getContext());
            recycler = null;
            return view;
        }
    }

```

DecorUtils.java
```
public static int getStratusBarHeight(Context context) {
        return getDimenSize(context, "status_bar_height");
    }

    private static int getDimenSize(Context context, String key) {
        int resourceId = context.getResources().getIdentifier(key, "dimen", "android");
        return resourceId > 0 ? context.getResources().getDimensionPixelSize(resourceId) : 0;
    }
```

**Comment by bgogetap** [comment]
Thanks, I will look into this tonight.

**Comment by bgogetap** [comment]
Ok, so this is only an issue if the RecyclerView is not positioned at 0,0, relative to the parent (hence the requirement for the parent to be a FrameLayout or CoordinatorLayout, for now).

I'm open to the idea of using the changes you've suggested remove that restriction, but it's not my immediate priority. It does look similar to how I'm thinking about handling padding on the RecyclerView in the `padding` branch, so I like the direction and will definitely look into it more in the future.

Otherwise, if you'd like to submit a PR once the `padding` branch changes get merged (will be version 0.2.0), feel free!
